src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseGroupServiceImpl.java

```diff
@@ -74,4 +88,27 @@ public void transitionCaseGroupStatus(final CaseGroup caseGroup, final CategoryD
 
   }
 
+  /**
+   * Use case ID to find case group
+   * Use case group to find collectionEx
+   * Then use CollEx service to find survey ID, then find all collexs for survey
+   * Then get all casge groups for the party ID where Collex ID is in list of collexs
+   * TODO: null checking
+   */
+  public List<CaseGroup> transitionOtherCaseGroups(final Category category, final CaseEvent caseEvent, final Case targetCase, Case newCase) throws CTPException {
+    CaseGroup caseGroup = caseGroupRepo.findOne(targetCase.getCaseGroupFK());
+    CollectionExerciseDTO collectionExercise = collectionExerciseSvcClientService
+            .getCollectionExercise(caseGroup.getCollectionExerciseId());
+    // fetch all the collection exercises for a survey
```

Comment:
Comments seem a bit over the top, your variable names seem pretty self documenting.